rfcs/0006-gtfs-canonical-routes.md

```diff
@@ -0,0 +1,125 @@
+times and shapes. If a canonical trip is similar enough to a real-world trip that are otherwise
+in the GTFS feed, then it will be made the representative trip of an existing route pattern. If no
+such trips are being run (such as due to a long-running station closure), then a new, canonical
+route pattern will be created, of which the canonical trip will be representative.
+
+# Motivation
+
+MBTA.com users are able to see line diagrams that match real-world state, including physical maps,
+even when stations are closed for a long duration.
+
+PIOs are able to create alerts for stations without any scheduled service, and T-Alerts subscribers
+are able to stay informed about the status of the station closure.
+
+# Guide-level explanation
+
+GTFS output includes additional, hardcoded data about where rail lines go to and where they can stop
+if all stations were open; these hardcoded data are independent of all scheduled service except that
+they may be tied to scheduled route patterns in some cases.
+
+This is represented as a service with ID "canonical", which is valid for the entire feed date range,
+but which has no active days of the week. The typicality of the service is tagged with a new value
+so it can be programmatically identified.
+
+All rail routes have two or more trips on this service, corresponding to the "canonical" patterns of
+service in each direction for any branches, as understood by stakeholders. Each such canonical trip
+has a route pattern, for which it is the representative trip, and a shape. If current service has
+trips following the same set and order of stops as the canonical trip, then the canonical trip will
+reference—and become the representative trip for—the current service's route pattern. If current
+service has no trips resembling the canonical trip, then a separate, canonical route pattern will be
+maintained in the GTFS output.
+
+# Reference-level explanation
+
+The canonical service appears in the following GTFS files in the following way:
+
+- **calendar.txt** includes a new `service_id` to envelop all of the canonical trips on all the
+  routes. `start_date` and `end_date` should reflect the feed's valid start/end dates. All other
+  fields should have a value of `0`.
+- **calendar_attributes.txt** includes the new `service_id`. `service_schedule_type` should be
+  `"Other"`, and `rating_start_date` and `rating_end_date` should reflect the feed's valid start/end
+  dates. `service_schedule_typicality` should be `6`, and the
+  [docs](https://github.com/mbta/gtfs-documentation/blob/master/reference/gtfs.md) should be updated
+  with this new kind of value.
+- **route_patterns.txt** includes new route patterns for routes/directions we want to hardcode when
+  the canonical trips (added in the trips file below) do not resemble any scheduled service. New
+  patterns in this file shall have a new `route_pattern_typicality` value of should be `6`, and
```

Comment:
I think this should be `5` (the highest current value is `4`).

Reply:
Great catch, thanks!